wbia/dtool/types.py

```diff
@@ -20,3 +33,147 @@
     dict: 'DICT',
     list: 'LIST',
 }
+
+
+class JSONCodeableType(UserDefinedType):
+
+    # Abstract properties
+    base_py_type = None
+    col_spec = None
+
+    def get_col_spec(self, **kw):
+        return self.col_spec
+
+    def bind_processor(self, dialect):
+        def process(value):
+            if value is None:
+                return value
+            else:
+                if isinstance(value, self.base_py_type):
+                    return json.dumps(value)
+                else:
+                    return value
+
+        return process
+
+    def result_processor(self, dialect, coltype):
+        def process(value):
+            if value is None:
+                return value
+            else:
+                if not isinstance(value, self.base_py_type):
+                    return json.loads(value)
+                else:
+                    return value
+
+        return process
+
+
+class NumPyPicklableType(UserDefinedType):
+
+    # Abstract properties
+    base_py_types = None
+    col_spec = None
+
+    def get_col_spec(self, **kw):
+        return self.col_spec
+
+    def bind_processor(self, dialect):
+        def process(value):
+            if value is None:
+                return value
+            else:
+                if isinstance(value, self.base_py_types):
+                    out = io.BytesIO()
+                    np.save(out, value)
+                    out.seek(0)
+                    return out.read()
+                else:
+                    return value
+
+        return process
+
+    def result_processor(self, dialect, coltype):
+        def process(value):
+            if value is None:
+                return value
+            else:
+                if not isinstance(value, self.base_py_types):
+                    out = io.BytesIO(value)
+                    out.seek(0)
+                    arr = np.load(out, allow_pickle=True)
+                    out.close()
+                    return arr
+                else:
+                    return value
+
+        return process
+
+
+class Dict(JSONCodeableType):
+    base_py_type = dict
+    col_spec = 'DICT'
+
+
+class Integer(TypeDecorator):
+    impl = SAInteger
+
+    def process_bind_param(self, value, dialect):
+        return int(value)
+
+
+class List(JSONCodeableType):
+    base_py_type = list
+    col_spec = 'LIST'
+
+
+class NDArray(NumPyPicklableType):
+    base_py_types = (np.ndarray,)
+    col_spec = 'NDARRAY'
+
+
+NP_NUMBER_TYPES = (
```

Comment:
nitpick: these are integers not numbers. Did you forget to put in floats? If not I'd rename to NP_INTEGER_TYPES

Reply:
Yes, we need to amend the name or the list of types to include np.float*